upload/src/addons/SV/ThreadReplyBanner/XF/Service/Thread/Creator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+
+namespace SV\ThreadReplyBanner\XF\Service\Thread;
+
+
+class Creator extends XFCP_Creator {
+
+	protected $bannerText = null;
+	protected $bannerActive = null;
+
+	public function setReplyBanner($text, $active) {
+		$this->bannerText = $text;
+		$this->bannerActive = $active;
+	}
+	
+	protected function _save() {
+		$thread = parent::_save();
+
+		if ($this->bannerText) {
+			$threadBanner = $this->em()->create('SV\ThreadReplyBanner:ThreadBanner');
+			$threadBanner->thread_id = $thread->thread_id;

```

Comment:
The above two lines can be simplified to; `$threadBanner = $thread->getRelationOrDefault('ThreadBanner');`